cli/deploy.go

```diff
@@ -0,0 +1,501 @@
+          image: quay.io/coreos/etcd:v2.2.2
+          imagePullPolicy: Always
+          args:
+            - --listen-client-urls
+            - http://0.0.0.0:2379
+            - --advertise-client-urls
+            - http://0.0.0.0:2379
+          ports:
+            - name: api
+              containerPort: 2379
+`
+
+var registry = `
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: registry
+  namespace: {{.Namespace}}
+  labels:
+    app: registry
+    type: storage
+spec:
+  selector:
+    app: registry
+    type: storage
+  ports:
+    - name: service
+      port: 5000
+      targetPort: 5000
+---
+kind: ReplicationController
+apiVersion: v1
+metadata:
+  name: registry
+  namespace: {{.Namespace}}
+  labels:
+    app: registry
+    type: storage
+spec:
+  replicas: 1
+  selector:
+    app: registry
+    type: storage
+  template:
+    metadata:
+      name: registry
+      namespace: {{.Namespace}}
+      labels:
+        app: registry
+        type: storage
+    spec:
+      containers:
+        - name: registry
+          image: registry:2
+          ports:
+            - name: service
+              containerPort: 5000
+
+`
+
+var kontinuousService = `
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: kontinuous
+  namespace: {{.Namespace}}
+  labels:
+    app: kontinuous
+    type: ci-cd
+spec:
+  type: LoadBalancer
+  selector:
+    app: kontinuous
+    type: ci-cd
+  ports:
+    - name: api
+      port: 8080
+      targetPort: 3005
+`
+
+var kontinuousRC = `
+---
+kind: ReplicationController
+apiVersion: v1
+metadata:
+  name: kontinuous
+  namespace: {{.Namespace}}
+  labels:
+    app: kontinuous
+    type: ci-cd
+spec:
+  replicas: 1
+  selector:
+    app: kontinuous
+    type: ci-cd
+  template:
+    metadata:
+      labels:
+        app: kontinuous
+        type: ci-cd
+    spec:
+      volumes:
+        - name: kontinuous-secrets
+          secret:
+            secretName: kontinuous-secrets
+      containers:
+        - name: kontinuous
+          image: quay.io/acaleph/kontinuous:latest
+          imagePullPolicy: Always
+          env:
+            - name: KV_ADDRESS
+              value: etcd:2379
+            - name: S3_URL
+              value: http://minio:9000
+            - name: KONTINUOUS_URL
+              value: http://{{.KontinuousIP}}:8080
+            - name: INTERNAL_REGISTRY
+              value: registry:5000
+          ports:
+            - name: api
+              containerPort: 3005
+          volumeMounts:
+            - mountPath: /.secret
+              name: kontinuous-secrets
+              readOnly: true
+`
+
+var dashboardSvc = `
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    service: kontinuous-ui
+    type: dashboard
+  name: kontinuous-ui
+  namespace: {{.Namespace}}
+spec:
+  ports:
+  - name: dashboard
+    nodePort: 30345
+    port: 5000
+    protocol: TCP
+    targetPort: 5000
+  selector:
+    app: kontinuous-ui
+    type: dashboard
+  type: LoadBalancer
```

Comment:
Should we make LB configurable for environments that might need to use Ingress instead? The end user can be responsible for creating the Ingress, we just toggle LB